tests/thread_manager_test.py

```diff
@@ -0,0 +1,22 @@
+"""Unit tests for thread_manager.ThreadManager."""
+
+import logging
+
+import discord
+import discord.abc
+import pytest
+
+from escape_roomba.context import Context
+from escape_roomba.thread_manager import ThreadManager
+
+
+@pytest.mark.asyncio
+async def test_thread_creation(mocker, make_context):
+    # Create a Context with a discord.Client-like Mock.
+    context = make_context(guild_count=2, channel_count=2, member_count=2)
+
+    # Create the ThreadManager and register its listeners.
+    ThreadManager(context)
+
```

Comment:
Interesting so this is where you're planning to test the creation of new channels?

are you going to mock/stub the discord API or are you going to create functions to check the channel count before and after?